test/integration/query/selects.js

```diff
@@ -1347,9 +1372,10 @@ module.exports = function (knex) {
               // mssql fails because it tires to rollback at the same time when update query is running
               // hopefully for share really works though...
               if (knex.client.driverName == 'mssql') {
-                expect(err.message).to.be.contain(
-                  "Can't rollback transaction. There is a request in progress"
-                );
+                // TODO: reconcile this behaviour under tedious
```

Comment:
Is this still relevant?

Reply:
I'm ...actually not sure what this is about, do you remember?

Reply:
Uncommenting out that test yields a `Requests can only be made in the LoggedIn state, not the SentClientRequest state` error message which is what tedious gives you when you try to make two requests at once.  Is that going to suffice?

Reply:
Yes, definitely!

Reply:
@kibertoad Sorry, not quite sure what you mean, and I'm a bit confused as to what's happening with this test.  I can simply update the error message in the test to make it pass, but that feels dirty. 

Although, it seems like a fuzzing test anyway that is checking for the presence of an error?

Reply:
I mean, since we are still throwing an error that is related to executing two requests at same time, sounds good enough, and just changing test assertion sounds sufficient. If previous error was thrown by node-mssql, I'm not sure if we should/can replicate throwing exact same error. I guess we could try-catch, wrap and rethrow error, but that sounds dirtier.

Reply:
The dialect transaction now throws an appropriate error when attempting to rollback a transaction that has a pending request, bringing it inline with how `node-mssql` handles it.  Tedious cannot ever roll back a deadlocked query.